tests/plugins/pytest_dut_monitor.py

```diff
@@ -0,0 +1,362 @@
+        self.run_channel = None
+        self._thread = threading.Thread(name="Connection tracker", target=self._track_connection)
+        self._thread.setDaemon(True)
+        self._thread.start()
+
+    def _track_connection(self):
+        """
+        @summary: Track network connectivity. Reestablish network connection in case of drop connection
+        """
+        while True:
+            try:
+                self.ssh.exec_command("ls", timeout=5)
+            except (paramiko.SSHException, AttributeError):
+                logger.warning("SSH connection droped")
+                logger.debug("Trying to reconnect...")
+                self.close()
+                try:
+                    self.init()
+                except Exception as err:
+                    logger.debug(repr(err))
+                else:
+                    if self.running:
+                        self.start()
+            else:
+                time.sleep(5)
+
+    def _upload_to_dut(self):
+        """
+        @summary: Upload 'dut_monitor.py' module to the DUT '/tmp' folder
+        """
+        logger.debug("Uploading file to the DUT...")
+        with self.ssh.open_sftp() as sftp:
+            sftp.put(os.path.join(os.path.split(__file__)[0], "dut_monitor.py"), DUT_MONITOR)
+
+    def init(self):
+        """
+        @summary: Connect to the DUT via SSH and authenticate to it.
+        """
+        logger.debug("Trying to establish connection ...")
+        self.ssh = paramiko.SSHClient()
+        self.ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+        self.ssh.connect(self.host, username=self.user, password=self.password, timeout=5)
+
+    def close(self):
+        """
+        @summary: Close this SSHClient and its underlying Transport
+        """
+        logger.debug("Close SSH connection with DUT")
+        self.ssh.close()
+
+    def exec_command(self, cmd, timeout=None):
+        """
+        @summary: Execute a command on the DUT and track possible connectivity issues. A new Channel is opened and the requested command is executed
+        """
+        try:
+            return self.ssh.exec_command(cmd, timeout=timeout, get_pty=True)
+        except Exception as err:
+            logger.warning("Broken connection - {}".format(repr(err)))
+            logger.warning("Skip command {}".format(cmd))
+            return (None, None, None)
+
+    def start(self):
+        """
+        @summary: Start HW resources monitoring on the DUT. Write obtained values to the following files on the DUT: DUT_CPU_LOG, DUT_RAM_LOG, DUT_HDD_LOG
+        """
+        self.running = True
+        self._upload_to_dut()
+        logger.debug("Start HW resuorces monitoring on the DUT...")
+
+        self.run_channel = self.ssh.get_transport().open_session()
+        self.run_channel.get_pty()
+        self.run_channel.settimeout(5)
+        # Start monitoring on DUT
+        self.run_channel.exec_command("python {} --start".format(DUT_MONITOR))
```

Comment:
How this survives when test performs reboot?

Reply:
There is mechanism to track network connectivity and restore monitoring script if network connectivity dropped.
It starts here:
self._thread = threading.Thread(name="Connection tracker", target=self._track_connection)
self._thread.setDaemon(True)
self._thread.start()